### DISCUSSION TITLE
Fabric version of `UIManagerModule.updateNodeSize` on Android?

### DISCUSSION THREAD

**Discussion by joemun** [reactwg/react-native-new-architecture#151] [Q&A] [answered]
Hi, curious if is there a fabric way for updating the size of a node in the React tree? akin to [UIManagerModule.updateNodeSize](https://github.com/facebook/react-native/blob/4e92f609af27505ddfa7b4ec38e88bed6c881d12/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/UIManagerModule.java#L383) ?

We have a use case where we are creating views natively in Android, but then need to update the React tree after getting the measurements. The behavior we are seeing is that the natively created view is not available in the shadow tree, and thus is exiting early without updating the size in React, in [UIImplementation.updateNodeSize](https://github.com/facebook/react-native/blob/4e92f609af27505ddfa7b4ec38e88bed6c881d12/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/UIImplementation.java#L198C8-L198C8).

Thank you!

**Comment by sammy-SC** [comment, marked as answer]
Hello @joemun,

yes, Fabric has an alternative to update the size of a node in the tree.
Fabric stores all this information in ShadowTree and you will have to subclass some classes to get this to work.
I recommend looking at how [Modal in Fabric](https://github.com/facebook/react-native/tree/main/packages/react-native/ReactCommon/react/renderer/components/modal) works. 
Through C++ state update, Android will tell Fabric about size. Then Fabric will apply this size to corresponding Yoga node and position everything accordingly.

**Reply by joemun** [reply]
Thank you for the response!

Curious if you all have also explored exposing this functionality in a more generic fashion, to update _any_ existing shadow node already in the tree? this is essentially what `updateNodeSize()` does given that you provide the view tag to update.

I do feel like others will encounter this need, and requiring end users to create new view components with backing C++ state, will be a high barrier.
